subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -0,0 +1,805 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud_codec.h>
+#include "nrf_cloud_mem.h"
+#include "nrf_cloud_codec_internal.h"
+
+LOG_MODULE_REGISTER(nrf_cloud_codec, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+int nrf_cloud_obj_input_decode(struct nrf_cloud_obj *const obj,
```

Comment:
what if the type hint was just the type of obj (`obj->type`) passed in?
if the type is undefined, it will try to decode the correct type.
if the type is defined, it will only try to decode that type.